k3k-kubelet/config.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"errors"
+	"os"
+
+	"gopkg.in/yaml.v2"
+)
+
+// config has all virtual-kubelet startup options
+type config struct {
+	ClusterName       string `yaml:"clusterName,omitempty"`
+	ClusterNamespace  string `yaml:"clusterNamespace,omitempty"`
+	NodeName          string `yaml:"nodeName,omitempty"`
+	Token             string `yaml:"token,omitempty"`
+	AgentHostname     string `yaml:"agentHostname,omitempty"`
+	HostConfigPath    string `yaml:"hostConfigPath,omitempty"`
+	VirtualConfigPath string `yaml:"virtualConfigPath,omitempty"`
+	KubeletPort       string `yaml:"kubeletPort,omitempty"`
+}
+
+func (c *config) unmarshalYAML(data []byte) error {
+	var conf config
+
+	if err := yaml.Unmarshal(data, &conf); err != nil {
+		return err
+	}
+
+	if c.ClusterName == "" {
+		c.ClusterName = conf.ClusterName
+	}
+	if c.ClusterNamespace == "" {
+		c.ClusterNamespace = conf.ClusterNamespace
+	}
+	if c.HostConfigPath == "" {
+		c.HostConfigPath = conf.HostConfigPath
+	}
+	if c.VirtualConfigPath == "" {
+		c.VirtualConfigPath = conf.VirtualConfigPath
+	}
+	if c.KubeletPort == "" {
+		c.KubeletPort = conf.KubeletPort
+	}
+	if c.AgentHostname == "" {
+		c.AgentHostname = conf.AgentHostname
+	}
+	if c.NodeName == "" {
+		c.NodeName = conf.NodeName
+	}
+	return nil
+}
```

Comment:
the problem is that we want the user to be able to mix and match for example, the user can set the --agent-hostname as a flag and at the same time add `nodeName` in the config file, if we just returned the unmarshalled config it will ignore all the flags sat on the command line